src/panel/register_masq.js

```diff
@@ -8,19 +8,57 @@ function RegisterMasqPanel() {
   this.store = new Store()
   this.panel = new Panel(this, RegisterMasqPanelView)
   this.isActive = false
+  this.registering = false
+  this.username = null
+  this.profileImage = null
+
+  this.isRegistered = false
+  this.store.isRegistered().then(async (b) => {
+    this.isRegistered = b
+    if (this.isRegistered) {
+      const { username, profileImage } = await this.store.getUserInfo()
+      this.username = username
+      this.profileImage = profileImage
+    }
+    this.panel.update()
+  })
 
   listen('register_panel__show', () => {
-    this.panel.animate(.25, '.register_masq_panel', {top : '100px'})
+    //this.panel.animate(.25, '.register_masq_panel', {top : '100px'})
   })
 }
 
-RegisterMasqPanel.prototype.register = function () {
-  this.store.register().then(() => {
-    this.panel.animate(.25, '.register_masq_panel', {top : '-300px'})
-  })
-  .catch((e) => {
+RegisterMasqPanel.prototype.register = async function () {
+  this.registering = true
+  this.panel.update()
+  try {
+    await this.store.register()
+    //this.panel.animate(.25, '.register_masq_panel', {top : '-300px'})
+    this.isRegistered = await this.store.isRegistered()
+    if (this.isRegistered) {
+      const { username, profileImage } = await this.store.getUserInfo()
+      this.username = username
+      this.profileImage = profileImage
+    }
+  } catch (e) {
     Error.sendOnce('register_masq', 'register', 'error registering masq', e)
-  })
+    this.isRegistered = await this.store.isRegistered()
+  }
+  this.registering = false
+  this.panel.update()
+}
+
+RegisterMasqPanel.prototype.unregister = async function () {
+  await this.store.unregister()
+  this.isRegistered = await this.store.isRegistered()
+  if (this.isRegistered) {
+    const { username, profileImage } = await this.store.getUserInfo()
+    this.username = username
+    this.profileImage = profileImage
+  }
+  window.location.reload()
```

Comment:
Oh waw the uge one

Reply:
:man_facepalming: I wanted to refresh the view and favorite list...
Changed to fire an event on login and logout caught by Favorite panel to refresh the list